src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Utilities/FileUtilities.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Globalization;
+using System.IO;
+
+namespace Microsoft.AspNetCore.Razor.Utilities;
+
+internal class FileUtilities
+{
+    /// <summary>
+    /// Generate a file path with adjacent to our input path that has the
+    /// correct file extension, using numbers to differentiate from
+    /// any collisions.
+    /// </summary>
+    /// <param name="path">The origin file path.</param>
+    /// <param name="extension">The desired file extension.</param>
+    /// <returns>A non-existent file path with a name in the desired format and a corresponding extension.</returns>
+    public static string GenerateUniquePath(string path, string extension)
```

Comment:
It's great that you added a helper method in a shared location, but when doing that it would be good to move the existing call to using it too. So in this case, `ExtractToCodeBehindCodeActionResolver` should be updated. Essentially, don't be afraid to update things that are strictly speaking outside the scope of the PR, as part of improving the overall code base.